partiql-logical-planner/src/call_defs.rs

```diff
@@ -441,6 +441,118 @@ fn function_call_def_cardinality() -> CallDef {
     }
 }
 
+fn function_call_def_extract() -> CallDef {
+    CallDef {
+        names: vec!["extract"],
+        overloads: vec![
+            CallSpec {
+                input: vec![
+                    CallSpecArg::Named("year".into()),
+                    CallSpecArg::Named("from".into()),
+                ],
+                output: Box::new(|mut args| {
+                    args.remove(0); // remove first default synthesized argument
```

Comment:
Where is the default _arg_ coming from?

Reply:
The default arg is from the preprocessor step that names the arguments:

https://github.com/partiql/partiql-lang-rust/blob/22379b9599d5551d6ec564052adc4ab4e77c624c/partiql-parser/src/preprocessor.rs#L92-L102

So for `EXTRACT(YEAR FROM x)`, the preprocessor would name the arguments: `YEAR: true` and `FROM: x`. Since we're looking at the `EXTRACT` function with a first `CallSpecArg` of "year" here, we can drop the first argument for `args`.

Reply:
Gotcha, maybe we can add the context to the comments for future?

Reply:
Sure, I can add that that arg comes from the parser preprocessor.